### PR TITLE
feat: added approval checks for multi-hop trades

### LINKED ISSUE #4762: ERC20 allowance checks for multi-hop trades
### Overview

We require implementations of ERC20 allowance checks so the UI knows whether to request an allowance transaction.

### References and additional details

Implement checks in the multi-hop trade execution flow to determine whether ERC20 allowance is adequate for each hop. 

### Acceptance Criteria

-  Multi-hop trade execution flow determines whether ERC20 allowance is adequate for each hop
- The allowance check is rechecked every 10 seconds so that changes on-chain are reflected in the app

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

2 hours

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Adds approval check for multi hop trades. Includes tiny fixes to asset selection buttons in multi hop trade input.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes  #4762

## Risk

No risk - not user facing

## Testing

This is tricky to test currently, but it might be possible if you can get a trade quote on a pair requiring erc20 approval.

### Engineering

See above

### Operations

No action required

### CODE REVIEW

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4759** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4759" target="_blank">Graphite</a> 
    * **PR #4761** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4761" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4761?utm_source=stack-comment).

File: src/components/MultiHopTrade/hooks/useIsApprovalNeeded.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    const allowanceOnChainCryptoBaseUnit = await getERC20Allowance({
+      web3,
+      erc20AllowanceAbi,
+      address: sellAssetContractAddress,
+      spender: allowanceContract,
+      from,
+    })
+
+    // TODO(woodenfurniture): This was pulled from the old implementation but we should check whether we should be
+    // including protocol feess in the sell asset here
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/hooks/useIsApprovalNeeded.tsx:49]: Nitpick for grep purposes
```suggestion
    // including protocol fees in the sell asset here
```

File: src/components/MultiHopTrade/hooks/useIsApprovalNeeded.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { usePoll } from 'hooks/usePoll/usePoll'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { bn } from 'lib/bignumber/bignumber'
+import type { TradeQuote } from 'lib/swapper/api'
+import { erc20AllowanceAbi } from 'lib/swapper/swappers/utils/abi/erc20Allowance-abi'
+import { getERC20Allowance } from 'lib/swapper/swappers/utils/helpers/helpers'
+import { getWeb3InstanceByChainId } from 'lib/web3-instance'
+
+const APPROVAL_CHECK_INTERVAL_MILLISECONDS = 10_000
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/hooks/useIsApprovalNeeded.tsx:12]: Something to potentially revisit in the future, for EVM chains like Gnosis, we could get away with smaller interval checks

**Review by gomesalexandre — APPROVED** [review]
I was unable to test this one as I couldn't get into a multi-hop trade requiring ERC-20 approval, but conceptually LGTM

**Review by gomesalexandre — APPROVED** [review]
Restamp after restack

**Pull request merged**